src/python/pants/backend/codegen/protobuf/lint/buf/format_rules.py

```diff
@@ -98,7 +99,7 @@ async def run_buf_format(
             output_files=request.files,
             description=f"Run buf format on {pluralize(len(request.files), 'file')}.",
             level=LogLevel.DEBUG,
-            env={"PATH": binary_shims.bin_directory},
+            env={"PATH": os.path.join("{chroot}", binary_shims.bin_directory)},
```

Comment:
This won't work with remote execution without a wrapper script. (This is a general problem of `{chroot}` not specific to `buf`.)

Reply:
🤔

Reply:
Coincidentally, this should be solvable in remote execution once https://github.com/pantsbuild/pants/pull/17290 lands because that adds a wrapper script for remote execution to support append-only caches in the remote environment. Then solving the PATH issue is just a matter of adding the additional shell code to substitute in the remote executor's input root path.

Reply:
```shell
PATH=${PATH//\{chroot\}/$(pwd)}
```